### DISCUSSION TITLE
Anti symmetric adjacency matrix using sklearn

### DISCUSSION THREAD

**Discussion by adhikarirsr** [pyg-team/pytorch_geometric#3449] [Q&A] [answered]
I have a code as follows:
```
xt=np.array([[0.753217, 
  0.0439285], [-0.827553, -0.244174], [0.0875135, -0.0413367], 
[-0.509302, 0.519792], [-0.0819656, 
  0.769458], [0.167709, -0.472054], [0.83912, -0.15233], [0.974581, 
  0.175885], [0.392318, 0.503732], [-0.196902, 0.265483]]);
adj = kneighbors_graph(xt, 2,metric='euclidean', mode='connectivity')
adj11 = adj.toarray()
(adj11==adj11.T).all()
=> False
```
I want to make an undirected graph. However ,sklearn gives me an antisymmetric matrix. Is there a way to get a symmetric adjacency matrix for the case above?

**Comment by rusty1s** [comment, marked as answer]
`k`-nearest neighbor graphs are always directed. You can make it undirected as follows:
```python
adj11 = adj11 + adj11.T
adj11[adj > 1] = 1.
```

**Reply by adhikarirsr** [reply]
Thanks @rusty1s 

That's what I was thinking too. 
There's a typo in the second line. It should be: `adj11[adj11 > 1] = 1` , right?

**Reply by rusty1s** [reply]
Yes :)
